### DISCUSSION TITLE
timeout Error using pymeasure for Agilent B1500

### DISCUSSION THREAD

**Discussion by Dianebonk** [pymeasure/pymeasure#1082] [Q&A] [answered]
Hello, 
I am trying to program B1500 in my lab and I get error while compiling code I get for example in this website :https://pymeasure.readthedocs.io/en/latest/api/instruments/agilent/agilentB1500.html#pymeasure.instruments.agilent.agilentB1500.AgilentB1500.query_modules 

```python
from pymeasure.instruments.agilent import AgilentB1500

# explicitly define r/w terminations; set sufficiently large timeout in milliseconds or None.
b1500=AgilentB1500("GPIB0::17::INSTR", read_termination='\r\n', write_termination='\r\n', timeout=600000)
```

```python
**ERROR:**  line 241, in findall
    return _compile(pattern, flags).findall(string)

TypeError: expected string or bytes-like object
```

it's this line that cause the problem, any ideas? Thanks

**Comment by BenediktBurger** [comment, marked as answer]
Does #1083 address your problem better?
